Add shell completion support

panos-span opened the pull request:
## Overview
This PR closes #52  and adds **shell completion support** to the `alexandria3k` CLI using the optional [`shtab`](https://pypi.org/project/shtab/) dependency.  

Completions are provided for:
- Global options (`--debug`, `--progress`, `--version`)
- Subcommands (`populate`, `query`, `process`, `download`, etc.)
- Data source names
- Process names
- Schema facilities
- File and path arguments

## Installation
Install with the recommended extra:
```bash
pip install 'alexandria3k[completion]'
```

Or install the dependency directly:
~~~bash
pip install shtab
~~~

## Usage
Generate a completion script:
~~~bash
a3k --print-completion SHELL   # SHELL = bash | zsh | tcsh
~~~

### Example
~~~console
$ a3k popu<TAB>
populate
$ a3k query -Q <TAB>
(file suggestions…)
~~~

## Notes
- Completions work for Bash, Zsh, and Tcsh.
- If `shtab` is missing, a helpful instruction is shown.
- Re-run `a3k --print-completion <shell>` after upgrading to refresh completions.

dspinellis commented:
Now that CI is fixed, please proceed as follows.

* [ ] Please clarify how `pip install shtab` can be a substitute for `pip install 'alexandria3k[completion]'` as you write in the Installation instructions.
* [ ] Consider making the installation an integral part when supported (e.g. Unix) or skipping it when not.  What is the best practice here? Adjust the a3k installation documentation accordingly.
* [ ] Please rebase your commits into a single one and `push -f` to check it again.

panos-span commented:
### 1. Clarification on `pip install shtab` vs `pip install 'alexandria3k[completion]'`

They are effectively substitutes because of how Python "extras" work.

*   **`pip install 'alexandria3k[completion]'`**: This command tells `pip` to install the `alexandria3k` package *plus* any dependencies listed in the `[project.optional-dependencies]` section under the key `completion`. In your `pyproject.toml`, that list contains only `shtab`.
*   **`pip install shtab`**: This manually installs the `shtab` package into your environment.

Since the code in `completion.py` simply checks if it can `import shtab`, it does not matter *how* the package arrived in the environment (via the extra or a manual install)—only that it is present.

### 2. Best Practice for Platform-Specific Dependencies

The best practice for features that are standard on some platforms (like shell completion on Unix) but optional on others is to use **environment markers** in your main dependency list. This makes the installation "integral" where supported and skips it where it's not, without requiring the user to remember extra flags.

I have applied this best practice to your project:

1.  **Updated `pyproject.toml`**:
    I moved `shtab` to the main `dependencies` list but added a condition so it only installs automatically on non-Windows systems.

    ```toml
    dependencies = [
      # ... other dependencies ...
      "shtab>=1.6; sys_platform != 'win32'",
    ]
    ```

2.  **Updated `docs/cli.rst`**:
    I adjusted the documentation to reflect that `shtab` is now installed automatically on non-Windows systems, while retaining instructions for Windows users or manual installation.

This ensures a seamless experience for Unix users while keeping the Windows installation lightweight by default.
